### PR TITLE
Use FormModalBody and ConfirmModalBody everywhere relevant

### LINKED ISSUE #1575: Make sure ModalBodyConfirm is used where appropriate. HARD 8
- [ ] Use ModalBodyConfirm
https://github.com/linode/manager/blob/master/src/domains/components/EditARecord.js#L111

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #1575, closes #2187

To check:

* Edit A Record
* Edit CNAME Record
* Edit MX Record
* Edit NS Record
* Edit SOA Record
* Edit SRV Record
* Edit TXT Record
* MasterZone
* Delete Domain
* Delete and reboot status dropdown
* Delete and reboot from mass actions
* Confirm restore backup
* Cancel backups
* Reset root password
* Add IP address
* Edit RDNS
* IP Address More Info
* Add Disk
* Edit Disk
* Delete Disk
* Delete Config
* Delete NodeBalancer
* Create Nodebalancer node
* Edit NodeBalancer node
* Delete NodeBalancer config
* Enable TFA
* Revoke Application
* Create Application
* Edit Application
* Delete Application
* Create PAT
* Edit PAT

### CODE REVIEW

File: src/linodes/linode/settings/advanced/components/AddDisk.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import _ from 'lodash';
```
Comment (mparke) [inline src/linodes/linode/settings/advanced/components/AddDisk.js]: is this a rename? or a rewrite?

Comment (eatonphil) [inline src/linodes/linode/settings/advanced/components/AddDisk.js]: This is just re-ordering alphabetically.

File: src/linodes/linode/settings/advanced/components/DiskPanel.js [hunk 2]
```diff
@@ -122,7 +93,7 @@ export class DiskPanel extends Component {
                 <small className="text-muted">Being deleted</small>}
                 {!this.poweredOff() || d.state === 'deleting' ? null : (
                   <div>
-                    <Button onClick={this.editAction(d)}>Edit</Button>
+                    <Button onClick={EditDisk.trigger(dispatch)}>Edit</Button>
```
Comment (mparke) [inline src/linodes/linode/settings/advanced/components/DiskPanel.js]: can we use these methods consistently, most other places trigger does not return a function, either rename or wrap in an anonymous function here

Comment (eatonphil) [inline src/linodes/linode/settings/advanced/components/DiskPanel.js]: Ah, whoops. Good catch. Thank you.

**Comment by mparke** [comment]
What are the benefits of defining static class methods here?

File: src/linodes/linode/settings/advanced/components/DiskPanel.js (hunk 2 above)
Comment (eatonphil) [inline src/linodes/linode/settings/advanced/components/DiskPanel.js]: Updated in latest commit.

**Comment by eatonphil** [comment]
@mparke merge conflicts fixed.

**Comment by eatonphil** [comment]
> What are the benefits of defining static class methods here?

Primarily it keeps the title attribute together. This is good because the same modal may be triggered from multiple places and we don't want to leave it up to the caller to decide the title each time. It needs to be a static method because it will be creating a new instances of the class itself.

**Comment by mparke** [comment]
Question about form submits here, I see some of them changed to async methods, why are the async additions needed?

**Comment by eatonphil** [comment]
@mparke I don't think anything changed in the places that _use_ the components but the components themselves make sure to first await the submit before doing anything afterwards.

File: components/dropdowns/Dropdown.js [hunk 3]
```diff
@@ -34,15 +34,15 @@ export default class Dropdown extends Component {
             <button
               type="button"
               key={item.name}
-              id={item.name.split(' ').join('-').toLowerCase()}
+              id={(item || '').name.split(' ').join('-').toLowerCase()}
```
Comment (mparke) [inline components/dropdowns/Dropdown.js]: `''.name.split` will throw,

was this intended to be `(item.name || '').split` like below? also if same calculation make a function

Comment (eatonphil) [inline components/dropdowns/Dropdown.js]: ugh. thanks

Comment (eatonphil) [inline components/dropdowns/Dropdown.js]: done

**Comment by mparke** [comment]
screen shot 2017-07-07 at 11 37 45 am

**Comment by mparke** [comment]
networking summary > more info warnings

screen shot 2017-07-07 at 11 40 11 am

**Comment by eatonphil** [comment]
@mparke thanks, made onCancel optional.

File: components/modals/FormModalBody.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        className={className}
+        onSubmit={this.onSubmit}
+        noSubmitEvent={noSubmitEvent}
+        title={this.props.title}
+        isEdit={this.props.isEdit}
+        type="Modal"
+      >
+        {React.isValidElement(children) ? children : <p>{children}</p>}
+        <div className="Modal-footer">
+          {noCancel ? null : <CancelButton disabled={loading} onClick={onCancel} />}
```
Comment (na3d) [inline components/modals/FormModalBody.js]: Should be:
`onClick={this.onCancel}`

As the props one is handled inside the this one.

Comment (eatonphil) [inline components/modals/FormModalBody.js]: done.

**Comment by mparke** [comment]
another warning that may be present here on clicking cancels:

```
Warning: Failed prop type: The prop `typeOfItem` is marked as required in `_type`, but its value is `undefined`.
    in _type`
```

File: components/modals/DeleteModalBody.js [hunk 5]
```diff
@@ -34,26 +36,29 @@ export default function DeleteModalBody(props) {
...
 
   return (
-    <ConfirmModalBody
+    <FormModalBody
       className="DeleteModalBody"
       buttonText="Delete"
       buttonDisabledText="Deleting"
-      onOk={() => {
-        onOk();
+      noSubmitEvent
```
Comment (abemassry) [inline components/modals/DeleteModalBody.js]: this uses the presence or lack thereof as a boolean value

File: components/modals/DeleteModalBody.js [hunk 6]
```diff
@@ -34,26 +36,29 @@ export default function DeleteModalBody(props) {
...
+      onSubmit={async () => {
+        await onSubmit();
         EmitEvent('modal:submit', 'Modal', 'delete', typeOfItem);
       }}
-      onCancel={() => {
-        onCancel();
+      noCancelEvent
```
Comment (abemassry) [inline components/modals/DeleteModalBody.js]: same here as noSubmitEvent

**Comment by na3d** [comment]
LGTM, ready for merge when @abemassry is happy,

**Comment by abemassry** [comment]
@eatonphil told me to hold off on this, so I'm going to keep my stranglehold on the merge button

File: components/dropdowns/Dropdown.js [hunk 7]
```diff
@@ -12,14 +14,41 @@ export default class Dropdown extends Component {
...
+
+    this.emitEvent(DROPDOWN_CLOSE, 'close');
+
     this.setState({ open: false });
   }
 
+  wrapClick(f) {
+    return (...args) => {
+      if (f !== this.open) {
+        this.emitEvent(DROPDOWN_CLICK, 'change');
```
Comment (na3d) [inline components/dropdowns/Dropdown.js]: Any possibility of getting more detail than change?

Comment (eatonphil) [inline components/dropdowns/Dropdown.js]: Yep, got a commit coming in a few minutes.

**Comment by eatonphil** [comment]
@na3d addressed. Thanks!

**Pull request merged**